Index: src/CodeBlocks_wx32_64.cbp
===================================================================
--- src/CodeBlocks_wx32_64.cbp	(revision 13254)
+++ src/CodeBlocks_wx32_64.cbp	(working copy)
@@ -2,14 +2,14 @@
 <CodeBlocks_project_file>
 	<FileVersion major="1" minor="6" />
 	<Project>
-		<Option title="Code::Blocks wx3.2.x (64 bit)" />
+		<Option title="Code::Blocks wx3.2.x noPCH (MSys2)" />
 		<Option default_target="src" />
 		<Option compiler="gcc" />
 		<Build>
 			<Target title="tinyXML">
-				<Option output="devel32_64/txml" prefix_auto="1" extension_auto="1" />
+				<Option output="devel32_msys2/txml" prefix_auto="1" extension_auto="1" />
 				<Option working_dir="" />
-				<Option object_output=".objs32_64" />
+				<Option object_output=".objs32_msys2" />
 				<Option type="2" />
 				<Option compiler="gcc" />
 				<Compiler>
@@ -19,7 +19,7 @@
 			</Target>
 			<Target title="AutoRevision">
 				<Option output="$(TARGET_OBJECT_DIR)/autorevision" prefix_auto="1" extension_auto="1" />
-				<Option object_output=".objs32_64" />
+				<Option object_output=".objs32_msys2" />
 				<Option type="1" />
 				<Option compiler="gcc" />
 				<Option parameters="-v +wx +int +t .. include/autorevision.h" />
@@ -33,9 +33,9 @@
 				</Linker>
 			</Target>
 			<Target title="ConsoleRunner">
-				<Option output="devel32_64/cb_console_runner" prefix_auto="1" extension_auto="1" />
+				<Option output="devel32_msys2/cb_console_runner" prefix_auto="1" extension_auto="1" />
 				<Option working_dir="tools/ConsoleRunner" />
-				<Option object_output=".objs32_64" />
+				<Option object_output=".objs32_msys2" />
 				<Option type="1" />
 				<Option compiler="gcc" />
 				<Option projectLinkerOptionsRelation="1" />
@@ -44,9 +44,9 @@
 				</Compiler>
 			</Target>
 			<Target title="Squirrel">
-				<Option output="devel32_64/squirrel" prefix_auto="1" extension_auto="1" />
+				<Option output="devel32_msys2/squirrel" prefix_auto="1" extension_auto="1" />
 				<Option working_dir="" />
-				<Option object_output=".objs32_64" />
+				<Option object_output=".objs32_msys2" />
 				<Option type="2" />
 				<Option compiler="gcc" />
 				<Compiler>
@@ -55,9 +55,9 @@
 				</Compiler>
 			</Target>
 			<Target title="scintilla">
-				<Option output="devel32_64/wxscintilla_cb" prefix_auto="1" extension_auto="1" />
+				<Option output="devel32_msys2/wxscintilla_cb" prefix_auto="1" extension_auto="1" />
 				<Option working_dir="" />
-				<Option object_output=".objs32_64" />
+				<Option object_output=".objs32_msys2" />
 				<Option type="2" />
 				<Option compiler="gcc" />
 				<Compiler>
@@ -82,9 +82,9 @@
 				</Linker>
 			</Target>
 			<Target title="sdk">
-				<Option output="devel32_64/codeblocks" prefix_auto="1" extension_auto="1" />
-				<Option working_dir="devel32_64" />
-				<Option object_output=".objs32_64" />
+				<Option output="devel32_msys2/codeblocks" prefix_auto="1" extension_auto="1" />
+				<Option working_dir="devel32_msys2" />
+				<Option object_output=".objs32_msys2" />
 				<Option type="3" />
 				<Option compiler="gcc" />
 				<Option parameters="--debug-log --no-dde --no-check-associations --multiple-instance --verbose -p debug" />
@@ -115,21 +115,24 @@
 					<Add library="txml" />
 					<Add library="wxscintilla_cb" />
 					<Add library="squirrel" />
+					<Add library="wx_$(WX_TOOLKIT)$(WX_SUFFIX)_xrc-$(WX_VERSION).dll" />
+					<Add library="wx_$(WX_TOOLKIT)$(WX_SUFFIX)_aui-$(WX_VERSION).dll" />
+					<Add library="wx_$(WX_TOOLKIT)$(WX_SUFFIX)_html-$(WX_VERSION).dll" />
+					<Add library="wx_$(WX_TOOLKIT)$(WX_SUFFIX)_core-$(WX_VERSION).dll" />
+					<Add library="wx_base$(WX_SUFFIX)_xml-$(WX_VERSION).dll" />
+					<Add library="wx_base$(WX_SUFFIX)_net-$(WX_VERSION).dll" />
 					<Add library="gdi32" />
 				</Linker>
 				<ExtraCommands>
 					<Add before="$(TARGET_OBJECT_DIR)\autorevision +wx +int +t .. include/autorevision.h" />
-					<Add after="cmd /c if not exist devel32_64\share\CodeBlocks mkdir devel32_64\share\CodeBlocks" />
-					<Add after="zip -jq9 devel32_64\share\CodeBlocks\manager_resources.zip sdk\resources\*.xrc" />
-					<Add after='cmd /c &quot;cd sdk\resources &amp; zip -0 -q ..\..\devel32_64\share\CodeBlocks\manager_resources.zip images\*.png images\8x8\*.png images\10x10\*.png images\12x12\*.png images\16x16\*.png images\20x20\*.png images\24x24\*.png images\28x28\*.png images\32x32\*.png images\40x40\*.png images\48x48\*.png images\56x56\*.png images\64x64\*.png images\svg\*.svg&quot;' />
 					<Mode after="always" />
 				</ExtraCommands>
 			</Target>
 			<Target title="src">
-				<Option output="devel32_64/codeblocks" prefix_auto="1" extension_auto="1" />
-				<Option working_dir="devel32_64" />
-				<Option object_output=".objs32_64" />
-				<Option external_deps="devel32_64/libcodeblocks.a;" />
+				<Option output="devel32_msys2/codeblocks" prefix_auto="1" extension_auto="1" />
+				<Option working_dir="devel32_msys2" />
+				<Option object_output=".objs32_msys2" />
+				<Option external_deps="devel32_msys2/libcodeblocks.a;" />
 				<Option type="0" />
 				<Option compiler="gcc" />
 				<Option parameters="--debug-log --no-dde --no-check-associations --multiple-instance --no-splash-screen --verbose -p debug" />
@@ -138,7 +141,6 @@
 					<Add option="-DBUILDING_PLUGIN" />
 					<Add directory="include" />
 					<Add directory="include/scripting/include" />
-					<Add directory="exchndl/win64/include" />
 				</Compiler>
 				<ResourceCompiler>
 					<Add option="-DcbDPI_AWARE_ON" />
@@ -147,7 +149,12 @@
 					<Add option="-Wl,--enable-auto-import" />
 					<Add option="-Wl,--no-undefined" />
 					<Add library="codeblocks" />
-					<Add library="exchndl.dll" />
+					<Add library="wx_$(WX_TOOLKIT)$(WX_SUFFIX)_propgrid-$(WX_VERSION).dll" />
+					<Add library="wx_$(WX_TOOLKIT)$(WX_SUFFIX)_xrc-$(WX_VERSION).dll" />
+					<Add library="wx_$(WX_TOOLKIT)$(WX_SUFFIX)_aui-$(WX_VERSION).dll" />
+					<Add library="wx_$(WX_TOOLKIT)$(WX_SUFFIX)_html-$(WX_VERSION).dll" />
+					<Add library="wx_$(WX_TOOLKIT)$(WX_SUFFIX)_core-$(WX_VERSION).dll" />
+					<Add library="exchndl" />
 					<Add library="shfolder" />
 					<Add library="kernel32" />
 					<Add library="user32" />
@@ -163,30 +170,19 @@
 					<Add library="rpcrt4" />
 					<Add library="advapi32" />
 					<Add library="wsock32" />
-					<Add directory="exchndl/win64/lib" />
 				</Linker>
-				<ExtraCommands>
-					<Add before="cmd /c if exist devel32_64\exchndl.dll del devel32_64\exchndl.dll" />
-					<Add after="cmd /c if not exist devel32_64\share\CodeBlocks mkdir devel32_64\share\CodeBlocks" />
-					<Add after="zip -jq9 devel32_64\share\CodeBlocks\resources.zip src\resources\*.xrc" />
-					<Add after="zip -jq9 devel32_64\share\CodeBlocks\start_here.zip src\resources\start_here\*.html src\resources\start_here\*.png" />
-					<Add after='cmd /c &quot;cd src\resources &amp; zip -0 -q ..\..\devel32_64\share\CodeBlocks\resources.zip images\*.png images\16x16\*.png&quot;' />
-					<Add after="cmd /c if not exist devel32_64\exchndl.dll if exist exchndl\win64\bin\*.dll xcopy /Y exchndl\win64\bin\*.dll devel32_64\" />
-					<Add after="cmd /c if not exist devel32_64\symsrv.yes if exist exchndl\win64\bin\symsrv.yes xcopy /Y exchndl\win64\bin\symsrv.yes devel32_64\" />
-					<Mode after="always" />
-				</ExtraCommands>
 			</Target>
 			<Target title="Abbreviations">
-				<Option output="devel32_64/share/CodeBlocks/plugins/abbreviations" prefix_auto="1" extension_auto="1" />
-				<Option working_dir="devel32_64" />
-				<Option object_output=".objs32_64" />
-				<Option external_deps="devel32_64/libcodeblocks.a;" />
+				<Option output="devel32_msys2/share/CodeBlocks/plugins/abbreviations" prefix_auto="1" extension_auto="1" />
+				<Option working_dir="devel32_msys2" />
+				<Option object_output=".objs32_msys2" />
+				<Option external_deps="devel32_msys2/libcodeblocks.a;" />
 				<Option type="3" />
 				<Option compiler="gcc" />
 				<Option parameters="--debug-log --no-dde --no-check-associations --multiple-instance --no-splash-screen --verbose -p debug" />
 				<Option host_application="./codeblocks.exe" />
 				<Option run_host_application_in_terminal="0" />
-				<Option projectLinkerOptionsRelation="1" />
+				<Option projectLinkerOptionsRelation="2" />
 				<Compiler>
 					<Add option="-DBUILDING_PLUGIN" />
 					<Add directory="include" />
@@ -199,25 +195,21 @@
 					<Add option="-Wl,--no-undefined" />
 					<Add option="-mthreads" />
 					<Add library="codeblocks" />
-					<Add library="wxmsw$(WX_VERSION)$(WX_SUFFIX)" />
+					<Add library="wx_$(WX_TOOLKIT)$(WX_SUFFIX)_xrc-$(WX_VERSION).dll" />
+					<Add library="wx_$(WX_TOOLKIT)$(WX_SUFFIX)_core-$(WX_VERSION).dll" />
 				</Linker>
-				<ExtraCommands>
-					<Add after="cmd /c if not exist devel32_64\share\CodeBlocks mkdir devel32_64\share\CodeBlocks" />
-					<Add after="zip -jq9 devel32_64\share\CodeBlocks\abbreviations.zip plugins\abbreviations\resources\manifest.xml plugins\abbreviations\resources\*.xrc" />
-					<Mode after="always" />
-				</ExtraCommands>
 			</Target>
 			<Target title="AStyle">
-				<Option output="devel32_64/share/CodeBlocks/plugins/astyle" prefix_auto="1" extension_auto="1" />
-				<Option working_dir="devel32_64" />
-				<Option object_output=".objs32_64" />
-				<Option external_deps="devel32_64/libcodeblocks.a;" />
+				<Option output="devel32_msys2/share/CodeBlocks/plugins/astyle" prefix_auto="1" extension_auto="1" />
+				<Option working_dir="devel32_msys2" />
+				<Option object_output=".objs32_msys2" />
+				<Option external_deps="devel32_msys2/libcodeblocks.a;" />
 				<Option type="3" />
 				<Option compiler="gcc" />
 				<Option parameters="--debug-log --no-dde --no-check-associations --multiple-instance --no-splash-screen --verbose -p debug" />
 				<Option host_application="./codeblocks.exe" />
 				<Option run_host_application_in_terminal="0" />
-				<Option projectLinkerOptionsRelation="1" />
+				<Option projectLinkerOptionsRelation="2" />
 				<Compiler>
 					<Add option="-DBUILDING_PLUGIN" />
 					<Add option="-DASTYLE_LIB" />
@@ -231,25 +223,21 @@
 					<Add option="-Wl,--no-undefined" />
 					<Add option="-mthreads" />
 					<Add library="codeblocks" />
-					<Add library="wxmsw$(WX_VERSION)$(WX_SUFFIX)" />
+					<Add library="wx_$(WX_TOOLKIT)$(WX_SUFFIX)_xrc-$(WX_VERSION).dll" />
+					<Add library="wx_$(WX_TOOLKIT)$(WX_SUFFIX)_core-$(WX_VERSION).dll" />
 				</Linker>
-				<ExtraCommands>
-					<Add after="cmd /c if not exist devel32_64\share\CodeBlocks mkdir devel32_64\share\CodeBlocks" />
-					<Add after="zip -jq9 devel32_64\share\CodeBlocks\astyle.zip plugins\astyle\resources\manifest.xml plugins\astyle\resources\*.xrc" />
-					<Mode after="always" />
-				</ExtraCommands>
 			</Target>
 			<Target title="Autosave">
-				<Option output="devel32_64/share/CodeBlocks/plugins/autosave" prefix_auto="1" extension_auto="1" />
-				<Option working_dir="devel32_64" />
-				<Option object_output=".objs32_64" />
-				<Option external_deps="devel32_64/libcodeblocks.a;" />
+				<Option output="devel32_msys2/share/CodeBlocks/plugins/autosave" prefix_auto="1" extension_auto="1" />
+				<Option working_dir="devel32_msys2" />
+				<Option object_output=".objs32_msys2" />
+				<Option external_deps="devel32_msys2/libcodeblocks.a;" />
 				<Option type="3" />
 				<Option compiler="gcc" />
 				<Option parameters="--debug-log --no-dde --no-check-associations --multiple-instance --no-splash-screen --verbose -p debug" />
 				<Option host_application="./codeblocks.exe" />
 				<Option run_host_application_in_terminal="0" />
-				<Option projectLinkerOptionsRelation="1" />
+				<Option projectLinkerOptionsRelation="2" />
 				<Compiler>
 					<Add option="-DBUILDING_PLUGIN" />
 					<Add directory="include" />
@@ -261,18 +249,14 @@
 					<Add option="-Wl,--no-undefined" />
 					<Add option="-mthreads" />
 					<Add library="codeblocks" />
-					<Add library="wxmsw$(WX_VERSION)$(WX_SUFFIX)" />
+					<Add library="wx_$(WX_TOOLKIT)$(WX_SUFFIX)_xrc-$(WX_VERSION).dll" />
+					<Add library="wx_$(WX_TOOLKIT)$(WX_SUFFIX)_core-$(WX_VERSION).dll" />
 				</Linker>
-				<ExtraCommands>
-					<Add after="cmd /c if not exist devel32_64\share\CodeBlocks mkdir devel32_64\share\CodeBlocks" />
-					<Add after="zip -jq9 devel32_64\share\CodeBlocks\autosave.zip plugins\autosave\manifest.xml plugins\autosave\*.xrc" />
-					<Mode after="always" />
-				</ExtraCommands>
 			</Target>
 			<Target title="Compiler depslib">
-				<Option output="devel32_64/depslib" prefix_auto="1" extension_auto="1" />
+				<Option output="devel32_msys2/depslib" prefix_auto="1" extension_auto="1" />
 				<Option working_dir="" />
-				<Option object_output=".objs32_64" />
+				<Option object_output=".objs32_msys2" />
 				<Option type="2" />
 				<Option compiler="gcc" />
 				<Compiler>
@@ -284,16 +268,16 @@
 				</Linker>
 			</Target>
 			<Target title="Compiler">
-				<Option output="devel32_64/share/CodeBlocks/plugins/compiler" prefix_auto="1" extension_auto="1" />
-				<Option working_dir="devel32_64" />
-				<Option object_output=".objs32_64" />
-				<Option external_deps="devel32_64/libcodeblocks.a;devel32_64/libdepslib.a;" />
+				<Option output="devel32_msys2/share/CodeBlocks/plugins/compiler" prefix_auto="1" extension_auto="1" />
+				<Option working_dir="devel32_msys2" />
+				<Option object_output=".objs32_msys2" />
+				<Option external_deps="devel32_msys2/libcodeblocks.a;devel32_msys2/libdepslib.a;" />
 				<Option type="3" />
 				<Option compiler="gcc" />
 				<Option parameters="--debug-log --no-dde --no-check-associations --multiple-instance --no-splash-screen --verbose -p debug" />
 				<Option host_application="./codeblocks.exe" />
 				<Option run_host_application_in_terminal="0" />
-				<Option projectLinkerOptionsRelation="1" />
+				<Option projectLinkerOptionsRelation="2" />
 				<Compiler>
 					<Add option="-DBUILDING_PLUGIN" />
 					<Add directory="include" />
@@ -307,29 +291,30 @@
 					<Add option="-Wl,--no-undefined" />
 					<Add option="-mthreads" />
 					<Add library="codeblocks" />
-					<Add library="wxmsw$(WX_VERSION)$(WX_SUFFIX)" />
 					<Add library="depslib" />
+					<Add library="wx_$(WX_TOOLKIT)$(WX_SUFFIX)_propgrid-$(WX_VERSION).dll" />
+					<Add library="wx_$(WX_TOOLKIT)$(WX_SUFFIX)_xrc-$(WX_VERSION).dll" />
+					<Add library="wx_$(WX_TOOLKIT)$(WX_SUFFIX)_core-$(WX_VERSION).dll" />
+					<Add library="wx_base$(WX_SUFFIX)_xml-$(WX_VERSION).dll" />
 				</Linker>
 				<ExtraCommands>
-					<Add after="cmd /c if not exist devel32_64\share\CodeBlocks mkdir devel32_64\share\CodeBlocks" />
-					<Add after="zip -jq9 devel32_64\share\CodeBlocks\compiler.zip plugins\compilergcc\resources\manifest.xml plugins\compilergcc\resources\*.xrc" />
-					<Add after='cmd /c &quot;cd plugins\compilergcc\resources &amp; zip -0 -q ..\..\..\devel32_64\share\CodeBlocks\compiler.zip images\*.png images\16x16\*.png images\20x20\*.png images\24x24\*.png images\28x28\*.png images\32x32\*.png images\40x40\*.png images\48x48\*.png images\56x56\*.png images\64x64\*.png images\svg\*.svg&quot;' />
-					<Add after="cmd /c if not exist devel32_64\share\CodeBlocks\compilers mkdir devel32_64\share\CodeBlocks\compilers" />
-					<Add after="cmd /c xcopy /D /Y plugins\compilergcc\resources\compilers\*.* devel32_64\share\CodeBlocks\compilers" />
+					<Add after="cmd /c if not exist devel32_msys2\share\CodeBlocks mkdir devel32_msys2\share\CodeBlocks" />
+					<Add after="cmd /c if not exist devel32_msys2\share\CodeBlocks\compilers mkdir devel32_msys2\share\CodeBlocks\compilers" />
+					<Add after="cmd /c xcopy /D /Y plugins\compilergcc\resources\compilers\*.* devel32_msys2\share\CodeBlocks\compilers" />
 					<Mode after="always" />
 				</ExtraCommands>
 			</Target>
 			<Target title="Debugger">
-				<Option output="devel32_64/share/CodeBlocks/plugins/debugger" prefix_auto="1" extension_auto="1" />
-				<Option working_dir="devel32_64" />
-				<Option object_output=".objs32_64" />
-				<Option external_deps="devel32_64/libcodeblocks.a;" />
+				<Option output="devel32_msys2/share/CodeBlocks/plugins/debugger" prefix_auto="1" extension_auto="1" />
+				<Option working_dir="devel32_msys2" />
+				<Option object_output=".objs32_msys2" />
+				<Option external_deps="devel32_msys2/libcodeblocks.a;" />
 				<Option type="3" />
 				<Option compiler="gcc" />
 				<Option parameters="--debug-log --no-dde --no-check-associations --multiple-instance --no-splash-screen --verbose -p debug" />
 				<Option host_application="./codeblocks.exe" />
 				<Option run_host_application_in_terminal="0" />
-				<Option projectLinkerOptionsRelation="1" />
+				<Option projectLinkerOptionsRelation="2" />
 				<Compiler>
 					<Add option="-DBUILDING_PLUGIN" />
 					<Add directory="include" />
@@ -341,25 +326,22 @@
 					<Add option="-Wl,--no-undefined" />
 					<Add option="-mthreads" />
 					<Add library="codeblocks" />
-					<Add library="wxmsw$(WX_VERSION)$(WX_SUFFIX)" />
+					<Add library="wx_$(WX_TOOLKIT)$(WX_SUFFIX)_xrc-$(WX_VERSION).dll" />
+					<Add library="wx_$(WX_TOOLKIT)$(WX_SUFFIX)_aui-$(WX_VERSION).dll" />
+					<Add library="wx_$(WX_TOOLKIT)$(WX_SUFFIX)_core-$(WX_VERSION).dll" />
 				</Linker>
-				<ExtraCommands>
-					<Add after="cmd /c if not exist devel32_64\share\CodeBlocks mkdir devel32_64\share\CodeBlocks" />
-					<Add after="zip -jq9 devel32_64\share\CodeBlocks\debugger.zip plugins\debuggergdb\resources\manifest.xml plugins\debuggergdb\resources\*.xrc" />
-					<Mode after="always" />
-				</ExtraCommands>
 			</Target>
 			<Target title="Code-completion">
-				<Option output="devel32_64/share/CodeBlocks/plugins/codecompletion" prefix_auto="1" extension_auto="1" />
-				<Option working_dir="devel32_64" />
-				<Option object_output=".objs32_64" />
-				<Option external_deps="devel32_64/libcodeblocks.a;" />
+				<Option output="devel32_msys2/share/CodeBlocks/plugins/codecompletion" prefix_auto="1" extension_auto="1" />
+				<Option working_dir="devel32_msys2" />
+				<Option object_output=".objs32_msys2" />
+				<Option external_deps="devel32_msys2/libcodeblocks.a;" />
 				<Option type="3" />
 				<Option compiler="gcc" />
 				<Option parameters="--debug-log --no-dde --no-check-associations --multiple-instance --no-splash-screen --verbose -p debug" />
 				<Option host_application="./codeblocks.exe" />
 				<Option run_host_application_in_terminal="0" />
-				<Option projectLinkerOptionsRelation="1" />
+				<Option projectLinkerOptionsRelation="2" />
 				<Compiler>
 					<Add option="-DBUILDING_PLUGIN" />
 					<Add directory="include" />
@@ -371,25 +353,22 @@
 					<Add option="-Wl,--no-undefined" />
 					<Add option="-mthreads" />
 					<Add library="codeblocks" />
-					<Add library="wxmsw$(WX_VERSION)$(WX_SUFFIX)" />
+					<Add library="wx_$(WX_TOOLKIT)$(WX_SUFFIX)_xrc-$(WX_VERSION).dll" />
+					<Add library="wx_$(WX_TOOLKIT)$(WX_SUFFIX)_aui-$(WX_VERSION).dll" />
+					<Add library="wx_$(WX_TOOLKIT)$(WX_SUFFIX)_core-$(WX_VERSION).dll" />
 				</Linker>
-				<ExtraCommands>
-					<Add after="cmd /c if not exist devel32_64\share\CodeBlocks mkdir devel32_64\share\CodeBlocks" />
-					<Add after="zip -jq9 devel32_64\share\CodeBlocks\codecompletion.zip plugins\codecompletion\resources\manifest.xml plugins\codecompletion\resources\*.xrc" />
-					<Mode after="always" />
-				</ExtraCommands>
 			</Target>
 			<Target title="Class wizard">
-				<Option output="devel32_64/share/CodeBlocks/plugins/classwizard" prefix_auto="1" extension_auto="1" />
-				<Option working_dir="devel32_64" />
-				<Option object_output=".objs32_64" />
-				<Option external_deps="devel32_64/libcodeblocks.a;" />
+				<Option output="devel32_msys2/share/CodeBlocks/plugins/classwizard" prefix_auto="1" extension_auto="1" />
+				<Option working_dir="devel32_msys2" />
+				<Option object_output=".objs32_msys2" />
+				<Option external_deps="devel32_msys2/libcodeblocks.a;" />
 				<Option type="3" />
 				<Option compiler="gcc" />
 				<Option parameters="--debug-log --no-dde --no-check-associations --multiple-instance --no-splash-screen --verbose -p debug" />
 				<Option host_application="./codeblocks.exe" />
 				<Option run_host_application_in_terminal="0" />
-				<Option projectLinkerOptionsRelation="1" />
+				<Option projectLinkerOptionsRelation="2" />
 				<Compiler>
 					<Add option="-DBUILDING_PLUGIN" />
 					<Add directory="include" />
@@ -401,25 +380,21 @@
 					<Add option="-Wl,--no-undefined" />
 					<Add option="-mthreads" />
 					<Add library="codeblocks" />
-					<Add library="wxmsw$(WX_VERSION)$(WX_SUFFIX)" />
+					<Add library="wx_$(WX_TOOLKIT)$(WX_SUFFIX)_xrc-$(WX_VERSION).dll" />
+					<Add library="wx_$(WX_TOOLKIT)$(WX_SUFFIX)_core-$(WX_VERSION).dll" />
 				</Linker>
-				<ExtraCommands>
-					<Add after="cmd /c if not exist devel32_64\share\CodeBlocks mkdir devel32_64\share\CodeBlocks" />
-					<Add after="zip -jq9 devel32_64\share\CodeBlocks\classwizard.zip plugins\classwizard\resources\manifest.xml plugins\classwizard\resources\*.xrc" />
-					<Mode after="always" />
-				</ExtraCommands>
 			</Target>
 			<Target title="Default MIME handler">
-				<Option output="devel32_64/share/CodeBlocks/plugins/defaultmimehandler" prefix_auto="1" extension_auto="1" />
-				<Option working_dir="devel32_64" />
-				<Option object_output=".objs32_64" />
-				<Option external_deps="devel32_64/libcodeblocks.a;" />
+				<Option output="devel32_msys2/share/CodeBlocks/plugins/defaultmimehandler" prefix_auto="1" extension_auto="1" />
+				<Option working_dir="devel32_msys2" />
+				<Option object_output=".objs32_msys2" />
+				<Option external_deps="devel32_msys2/libcodeblocks.a;" />
 				<Option type="3" />
 				<Option compiler="gcc" />
 				<Option parameters="--debug-log --no-dde --no-check-associations --multiple-instance --no-splash-screen --verbose -p debug" />
 				<Option host_application="./codeblocks.exe" />
 				<Option run_host_application_in_terminal="0" />
-				<Option projectLinkerOptionsRelation="1" />
+				<Option projectLinkerOptionsRelation="2" />
 				<Compiler>
 					<Add option="-DBUILDING_PLUGIN" />
 					<Add directory="include" />
@@ -431,25 +406,22 @@
 					<Add option="-Wl,--no-undefined" />
 					<Add option="-mthreads" />
 					<Add library="codeblocks" />
-					<Add library="wxmsw$(WX_VERSION)$(WX_SUFFIX)" />
+					<Add library="wx_$(WX_TOOLKIT)$(WX_SUFFIX)_xrc-$(WX_VERSION).dll" />
+					<Add library="wx_$(WX_TOOLKIT)$(WX_SUFFIX)_html-$(WX_VERSION).dll" />
+					<Add library="wx_$(WX_TOOLKIT)$(WX_SUFFIX)_core-$(WX_VERSION).dll" />
 				</Linker>
-				<ExtraCommands>
-					<Add after="cmd /c if not exist devel32_64\share\CodeBlocks mkdir devel32_64\share\CodeBlocks" />
-					<Add after="zip -jq9 devel32_64\share\CodeBlocks\defaultmimehandler.zip plugins\defaultmimehandler\resources\manifest.xml plugins\defaultmimehandler\resources\*.xrc" />
-					<Mode after="always" />
-				</ExtraCommands>
 			</Target>
 			<Target title="Occurrences Highlighting">
-				<Option output="devel32_64/share/CodeBlocks/plugins/OccurrencesHighlighting" prefix_auto="1" extension_auto="1" />
-				<Option working_dir="devel32_64" />
-				<Option object_output=".objs32_64" />
-				<Option external_deps="devel32_64/libcodeblocks.a;" />
+				<Option output="devel32_msys2/share/CodeBlocks/plugins/OccurrencesHighlighting" prefix_auto="1" extension_auto="1" />
+				<Option working_dir="devel32_msys2" />
+				<Option object_output=".objs32_msys2" />
+				<Option external_deps="devel32_msys2/libcodeblocks.a;" />
 				<Option type="3" />
 				<Option compiler="gcc" />
 				<Option parameters="--debug-log --no-dde --no-check-associations --multiple-instance --no-splash-screen --verbose -p debug" />
 				<Option host_application="./codeblocks.exe" />
 				<Option run_host_application_in_terminal="0" />
-				<Option projectLinkerOptionsRelation="1" />
+				<Option projectLinkerOptionsRelation="2" />
 				<Compiler>
 					<Add option="-DBUILDING_PLUGIN" />
 					<Add directory="include" />
@@ -461,25 +433,21 @@
 					<Add option="-Wl,--no-undefined" />
 					<Add option="-mthreads" />
 					<Add library="codeblocks" />
-					<Add library="wxmsw$(WX_VERSION)$(WX_SUFFIX)" />
+					<Add library="wx_$(WX_TOOLKIT)$(WX_SUFFIX)_xrc-$(WX_VERSION).dll" />
+					<Add library="wx_$(WX_TOOLKIT)$(WX_SUFFIX)_core-$(WX_VERSION).dll" />
 				</Linker>
-				<ExtraCommands>
-					<Add after="cmd /c if not exist devel32_64\share\CodeBlocks mkdir devel32_64\share\CodeBlocks" />
-					<Add after="zip -j9 devel32_64\share\CodeBlocks\OccurrencesHighlighting.zip plugins\occurrenceshighlighting\resources\manifest.xml plugins\occurrenceshighlighting\resources\*.xrc" />
-					<Mode after="always" />
-				</ExtraCommands>
 			</Target>
 			<Target title="Open files list">
-				<Option output="devel32_64/share/CodeBlocks/plugins/openfileslist" prefix_auto="1" extension_auto="1" />
-				<Option working_dir="devel32_64" />
-				<Option object_output=".objs32_64" />
-				<Option external_deps="devel32_64/libcodeblocks.a;" />
+				<Option output="devel32_msys2/share/CodeBlocks/plugins/openfileslist" prefix_auto="1" extension_auto="1" />
+				<Option working_dir="devel32_msys2" />
+				<Option object_output=".objs32_msys2" />
+				<Option external_deps="devel32_msys2/libcodeblocks.a;" />
 				<Option type="3" />
 				<Option compiler="gcc" />
 				<Option parameters="--debug-log --no-dde --no-check-associations --multiple-instance --no-splash-screen --verbose -p debug" />
 				<Option host_application="./codeblocks.exe" />
 				<Option run_host_application_in_terminal="0" />
-				<Option projectLinkerOptionsRelation="1" />
+				<Option projectLinkerOptionsRelation="2" />
 				<Compiler>
 					<Add option="-DBUILDING_PLUGIN" />
 					<Add directory="include" />
@@ -491,25 +459,20 @@
 					<Add option="-Wl,--no-undefined" />
 					<Add option="-mthreads" />
 					<Add library="codeblocks" />
-					<Add library="wxmsw$(WX_VERSION)$(WX_SUFFIX)" />
+					<Add library="wx_$(WX_TOOLKIT)$(WX_SUFFIX)_core-$(WX_VERSION).dll" />
 				</Linker>
-				<ExtraCommands>
-					<Add after="cmd /c if not exist devel32_64\share\CodeBlocks mkdir devel32_64\share\CodeBlocks" />
-					<Add after="zip -jq9 devel32_64\share\CodeBlocks\openfileslist.zip plugins\openfileslist\manifest.xml" />
-					<Mode after="always" />
-				</ExtraCommands>
 			</Target>
 			<Target title="Projects-workspaces importer">
-				<Option output="devel32_64/share/CodeBlocks/plugins/projectsimporter" prefix_auto="1" extension_auto="1" />
-				<Option working_dir="devel32_64" />
-				<Option object_output=".objs32_64" />
-				<Option external_deps="devel32_64/libcodeblocks.a;" />
+				<Option output="devel32_msys2/share/CodeBlocks/plugins/projectsimporter" prefix_auto="1" extension_auto="1" />
+				<Option working_dir="devel32_msys2" />
+				<Option object_output=".objs32_msys2" />
+				<Option external_deps="devel32_msys2/libcodeblocks.a;" />
 				<Option type="3" />
 				<Option compiler="gcc" />
 				<Option parameters="--debug-log --no-dde --no-check-associations --multiple-instance --no-splash-screen --verbose -p debug" />
 				<Option host_application="./codeblocks.exe" />
 				<Option run_host_application_in_terminal="0" />
-				<Option projectLinkerOptionsRelation="1" />
+				<Option projectLinkerOptionsRelation="2" />
 				<Compiler>
 					<Add option="-DBUILDING_PLUGIN" />
 					<Add directory="include" />
@@ -521,25 +484,20 @@
 					<Add option="-Wl,--no-undefined" />
 					<Add option="-mthreads" />
 					<Add library="codeblocks" />
-					<Add library="wxmsw$(WX_VERSION)$(WX_SUFFIX)" />
+					<Add library="wx_$(WX_TOOLKIT)$(WX_SUFFIX)_core-$(WX_VERSION).dll" />
 				</Linker>
-				<ExtraCommands>
-					<Add after="cmd /c if not exist devel32_64\share\CodeBlocks mkdir devel32_64\share\CodeBlocks" />
-					<Add after="zip -jq9 devel32_64\share\CodeBlocks\projectsimporter.zip plugins\projectsimporter\resources\manifest.xml plugins\projectsimporter\resources\*.xrc" />
-					<Mode after="always" />
-				</ExtraCommands>
 			</Target>
 			<Target title="Scripted wizard">
-				<Option output="devel32_64/share/CodeBlocks/plugins/scriptedwizard" prefix_auto="1" extension_auto="1" />
-				<Option working_dir="devel32_64" />
-				<Option object_output=".objs32_64" />
-				<Option external_deps="devel32_64/libcodeblocks.a;" />
+				<Option output="devel32_msys2/share/CodeBlocks/plugins/scriptedwizard" prefix_auto="1" extension_auto="1" />
+				<Option working_dir="devel32_msys2" />
+				<Option object_output=".objs32_msys2" />
+				<Option external_deps="devel32_msys2/libcodeblocks.a;" />
 				<Option type="3" />
 				<Option compiler="gcc" />
 				<Option parameters="--debug-log --no-dde --no-check-associations --multiple-instance --no-splash-screen --verbose -p debug" />
 				<Option host_application="./codeblocks.exe" />
 				<Option run_host_application_in_terminal="0" />
-				<Option projectLinkerOptionsRelation="1" />
+				<Option projectLinkerOptionsRelation="2" />
 				<Compiler>
 					<Add option="-DBUILDING_PLUGIN" />
 					<Add directory="include" />
@@ -552,25 +510,21 @@
 					<Add option="-Wl,--no-undefined" />
 					<Add option="-mthreads" />
 					<Add library="codeblocks" />
-					<Add library="wxmsw$(WX_VERSION)$(WX_SUFFIX)" />
+					<Add library="wx_$(WX_TOOLKIT)$(WX_SUFFIX)_xrc-$(WX_VERSION).dll" />
+					<Add library="wx_$(WX_TOOLKIT)$(WX_SUFFIX)_core-$(WX_VERSION).dll" />
 				</Linker>
-				<ExtraCommands>
-					<Add after="cmd /c if not exist devel32_64\share\CodeBlocks mkdir devel32_64\share\CodeBlocks" />
-					<Add after="zip -jq9 devel32_64\share\CodeBlocks\scriptedwizard.zip plugins\scriptedwizard\resources\manifest.xml" />
-					<Mode after="always" />
-				</ExtraCommands>
 			</Target>
 			<Target title="To-do">
-				<Option output="devel32_64/share/CodeBlocks/plugins/todo" prefix_auto="1" extension_auto="1" />
-				<Option working_dir="devel32_64" />
-				<Option object_output=".objs32_64" />
-				<Option external_deps="devel32_64/libcodeblocks.a;" />
+				<Option output="devel32_msys2/share/CodeBlocks/plugins/todo" prefix_auto="1" extension_auto="1" />
+				<Option working_dir="devel32_msys2" />
+				<Option object_output=".objs32_msys2" />
+				<Option external_deps="devel32_msys2/libcodeblocks.a;" />
 				<Option type="3" />
 				<Option compiler="gcc" />
 				<Option parameters="--debug-log --no-dde --no-check-associations --multiple-instance --no-splash-screen --verbose -p debug" />
 				<Option host_application="./codeblocks.exe" />
 				<Option run_host_application_in_terminal="0" />
-				<Option projectLinkerOptionsRelation="1" />
+				<Option projectLinkerOptionsRelation="2" />
 				<Compiler>
 					<Add option="-DBUILDING_PLUGIN" />
 					<Add directory="include" />
@@ -582,25 +536,21 @@
 					<Add option="-Wl,--no-undefined" />
 					<Add option="-mthreads" />
 					<Add library="codeblocks" />
-					<Add library="wxmsw$(WX_VERSION)$(WX_SUFFIX)" />
+					<Add library="wx_$(WX_TOOLKIT)$(WX_SUFFIX)_xrc-$(WX_VERSION).dll" />
+					<Add library="wx_$(WX_TOOLKIT)$(WX_SUFFIX)_core-$(WX_VERSION).dll" />
 				</Linker>
-				<ExtraCommands>
-					<Add after="cmd /c if not exist devel32_64\share\CodeBlocks mkdir devel32_64\share\CodeBlocks" />
-					<Add after="zip -jq9 devel32_64\share\CodeBlocks\todo.zip plugins\todo\resources\manifest.xml plugins\todo\resources\*.xrc" />
-					<Mode after="always" />
-				</ExtraCommands>
 			</Target>
 			<Target title="XP look &amp; feel">
-				<Option output="devel32_64/share/CodeBlocks/plugins/xpmanifest" prefix_auto="1" extension_auto="1" />
-				<Option working_dir="devel32_64" />
-				<Option object_output=".objs32_64" />
-				<Option external_deps="devel32_64/libcodeblocks.a;" />
+				<Option output="devel32_msys2/share/CodeBlocks/plugins/xpmanifest" prefix_auto="1" extension_auto="1" />
+				<Option working_dir="devel32_msys2" />
+				<Option object_output=".objs32_msys2" />
+				<Option external_deps="devel32_msys2/libcodeblocks.a;" />
 				<Option type="3" />
 				<Option compiler="gcc" />
 				<Option parameters="--debug-log --no-dde --no-check-associations --multiple-instance --no-splash-screen --verbose -p debug" />
 				<Option host_application="./codeblocks.exe" />
 				<Option run_host_application_in_terminal="0" />
-				<Option projectLinkerOptionsRelation="1" />
+				<Option projectLinkerOptionsRelation="2" />
 				<Compiler>
 					<Add option="-DBUILDING_PLUGIN" />
 					<Add directory="include" />
@@ -612,18 +562,17 @@
 					<Add option="-Wl,--no-undefined" />
 					<Add option="-mthreads" />
 					<Add library="codeblocks" />
-					<Add library="wxmsw$(WX_VERSION)$(WX_SUFFIX)" />
 				</Linker>
 				<ExtraCommands>
-					<Add after="cmd /c if not exist devel32_64\share\CodeBlocks mkdir devel32_64\share\CodeBlocks" />
-					<Add after="zip -jq9 devel32_64\share\CodeBlocks\xpmanifest.zip plugins\xpmanifest\manifest.xml" />
+					<Add after="cmd /c if not exist devel32_msys2\share\CodeBlocks mkdir devel32_msys2\share\CodeBlocks" />
+					<Add after="zip -jq9 devel32_msys2\share\CodeBlocks\xpmanifest.zip plugins\xpmanifest\manifest.xml" />
 					<Mode after="always" />
 				</ExtraCommands>
 			</Target>
 			<Environment>
-				<Variable name="WX_CFG" value="" />
 				<Variable name="WX_SUFFIX" value="u" />
-				<Variable name="WX_VERSION" value="32" />
+				<Variable name="WX_TOOLKIT" value="msw" />
+				<Variable name="WX_VERSION" value="3.2" />
 			</Environment>
 		</Build>
 		<VirtualTargets>
@@ -634,7 +583,6 @@
 		<Compiler>
 			<Add option="-Wall" />
 			<Add option="-std=gnu++11" />
-			<Add option="-m64" />
 			<Add option="$(#CB_RELEASE_TYPE)" />
 			<Add option="-pipe" />
 			<Add option="-mthreads" />
@@ -645,23 +593,19 @@
 			<Add option="-D__WXMSW__" />
 			<Add option="-DWXUSINGDLL" />
 			<Add option="-DcbDEBUG" />
-			<Add option="-DCB_PRECOMP" />
+			<Add option="-DNOPCH" />
 			<Add option="-DwxUSE_UNICODE" />
-			<Add option="-D_WIN64" />
-			<Add directory="$(#WX32_64.include)" />
-			<Add directory="$(#WX32_64.lib)/gcc_dll$(WX_CFG)/msw$(WX_SUFFIX)" />
 			<Add directory="sdk/wxscintilla/include" />
 			<Add directory="include/tinyxml" />
+			<Add directory="$(TARGET_COMPILER_DIR)lib/wx/include/$(WX_TOOLKIT)-unicode-$(WX_VERSION)" />
+			<Add directory="$(TARGET_COMPILER_DIR)include/wx-$(WX_VERSION)" />
 		</Compiler>
 		<ResourceCompiler>
-			<Add directory="$(#WX32_64.include)" />
-			<Add directory="$(#WX32_64)/lib/gcc_dll$(WX_CFG)/msw$(WX_SUFFIX)" />
+			<Add directory="$(TARGET_COMPILER_DIR)include/wx-$(WX_VERSION)" />
 		</ResourceCompiler>
 		<Linker>
-			<Add option="-m64" />
-			<Add library="wxmsw$(WX_VERSION)$(WX_SUFFIX)" />
-			<Add directory="$(#WX32_64.lib)/gcc_dll$(WX_CFG)" />
-			<Add directory="devel32_64" />
+			<Add library="wx_base$(WX_SUFFIX)-$(WX_VERSION).dll" />
+			<Add directory="devel32_msys2" />
 		</Linker>
 		<Unit filename="base/tinyxml/tinystr.cpp">
 			<Option target="tinyXML" />
@@ -988,7 +932,6 @@
 			<Option target="sdk" />
 		</Unit>
 		<Unit filename="include/sdk.h">
-			<Option compile="1" />
 			<Option weight="1" />
 			<Option target="src" />
 		</Unit>
@@ -999,7 +942,6 @@
 			<Option target="sdk" />
 		</Unit>
 		<Unit filename="include/sdk_precomp.h">
-			<Option compile="1" />
 			<Option weight="0" />
 			<Option target="sdk" />
 		</Unit>
